debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorTask.java

```diff
@@ -0,0 +1,177 @@
+                .loggingContextSupplier(() -> taskContext.configureLoggingContext(CONTEXT_NAME))
+                .build();
+
+        errorHandler = new ErrorHandler(OracleConnector.class, connectorConfig.getLogicalName(), queue, this::cleanupResources);
+        TopicSelector topicSelector = OracleTopicSelector.defaultSelector(connectorConfig.getLogicalName());
+
+        Configuration jdbcConfig = config.subset("database.", true);
+
+        jdbcConnection = new OracleConnection(jdbcConfig, new OracleConnectionFactory());
+        SchemaNameAdjuster schemaNameAdjuster = SchemaNameAdjuster.create(LOGGER);
+
+        this.schema = new OracleDatabaseSchema(connectorConfig, schemaNameAdjuster, topicSelector);
+
+        OracleOffsetContext previousOffset = getPreviousOffset(connectorConfig);
+        if (previousOffset != null) {
+            schema.recover(previousOffset);
+        }
+
+        EventDispatcher dispatcher = new EventDispatcher(topicSelector, schema, queue);
+
+        coordinator = new ChangeEventSourceCoordinator(
+                previousOffset,
+                errorHandler,
+                OracleConnector.class,
+                connectorConfig.getLogicalName(),
+                new OracleChangeEventSourceFactory(connectorConfig, jdbcConnection, errorHandler, dispatcher, clock, schema)
+        );
+
+        coordinator.start();
+    }
+
+    private OracleOffsetContext getPreviousOffset(OracleConnectorConfig connectorConfig) {
+        OracleOffsetContext offsetContext = new OracleOffsetContext(connectorConfig.getLogicalName());
+
+        Map<String, Object> previousOffset = context.offsetStorageReader()
+                .offsets(Collections.singleton(offsetContext.getPartition()))
+                .get(offsetContext.getPartition());
+
+        if (previousOffset != null) {
+            long scn = (long) previousOffset.get(SourceInfo.SCN_KEY);
+            offsetContext.setScn(scn);
+            LOGGER.info("Found previous offset {}", offsetContext);
+
+            return offsetContext;
+        }
+
+        return null;
+    }
+
+    @Override
+    public List<SourceRecord> poll() throws InterruptedException {
+        // TODO
+        List records = queue.poll();
+
+        return ((List<DataChangeEvent>)records).stream()
+            .map(DataChangeEvent::getRecord)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public void stop() {
+        cleanupResources();
+    }
+
+    private void cleanupResources() {
+        if (!state.compareAndSet(State.RUNNING, State.STOPPED)) {
+            LOGGER.info("Connector has already been stopped");
+            return;
+        }
+
+        try {
+            if (coordinator != null) {
+                coordinator.stop();
+            }
+        }
+        catch (InterruptedException e) {
+            Thread.interrupted();
+            LOGGER.error("Interrupted while stopping coordinator", e);
+        }
+
+        try {
```

Comment:
This is called from error handling thread in case of error effectivelly having task to terminate its own threadpool.